src/extension.ts

```diff
@@ -63,12 +82,29 @@ export function activate(context: ExtensionContext) {
     });
 
     workspace.onDidSaveTextDocument(doc => {
-        if (ameba && ameba.config.onSave) ameba.execute(doc);
+        if (ameba && ameba.config.onSave && doc.languageId === 'crystal' && !doc.isUntitled && doc.uri.scheme === 'file') {
```

Comment:
AFAIU that's a bugfix change. If so, please mention it in the PR title/description.

Reply:
Same as other location, it's to ensure the log isn't shown for invalid files

Reply:
It's not about log but the added `doc.languageId === 'crystal' && !doc.isUntitled && doc.uri.scheme === 'file'` conditions, which weren't there before.

Aside of logging this callback executes `ameba.execute(doc)`, so this is a substantial change as I understand it.

Reply:
I see now, thanks for the explanation.